numalogic/registry/redis_registry.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import time
+from typing import Optional, Sequence, Dict, Any
+
+from redis.client import Redis
+from redis.exceptions import RedisError
+
+from numalogic.registry import ArtifactManager, ArtifactData
+from numalogic.registry.serialize import loads, dumps
+from numalogic.tools.exceptions import ModelKeyNotFound
+from numalogic.tools.types import Artifact
+
+_LOGGER = logging.getLogger()
+
+
+class RedisRegistry(ArtifactManager):
+    """
+    Model saving and loading using Redis Registry.
+    Args:
+        client: Take in the reids client already established/created
+        ttl: Total Time to Live for the key when saving in redis (dafault = 5000)
+
+    Examples
+    --------
+    >>> import redis
+    >>> from numalogic.models.autoencoder.variants import VanillaAE
+    >>> from numalogic.registry.redis_registry import RedisRegistry
+    >>> r = redis.StrictRedis(host='127.0.0.1', port=6379)
```

Comment:
Understand this is an example. but want to double check if we use FQDN to connect to Redis. Otherwise, what happens when the Redis IP changes during pod rebalance?

Reply:
We will not be using IP, we will be using domain names